exporter/elasticsearchexporter/exporter.go

```diff
@@ -189,7 +190,7 @@ func (e *elasticsearchExporter) pushMetricsData(
 	}
 	defer session.End()
 
-	groupedDataPointsByIndex := make(map[elasticsearch.Index]map[uint32]*dataPointsGroup)
+	groupedDataPointsByIndex := make(map[elasticsearch.Index]map[string]map[uint32]*dataPointsGroup)
```

Comment:
This is tricky. I'm not sure if this is the right behavior for TSDB. If there exist 2 data points that need to go through different pipelines, should they be grouped together?
- If they are grouped together, we cannot specify 2 different pipelines
- If they are not grouped together, it will result in duplicates and data loss.
cc @felixbarny

Reply:
If we really need the dynamic pipeline to work with metrics, we need to accept and document that there may be TSDB conflicts and data loss.